crates/node/builder/src/launch/engine.rs

```diff
@@ -344,15 +349,52 @@ where
             let mut res = Ok(());
 
             // advance the chain and await payloads built locally to add into the engine api tree handler to prevent re-execution if that block is received as payload from the CL
+            let mut new_payload_responses = futures::stream::FuturesUnordered::default();
             loop {
                 tokio::select! {
                     payload = built_payloads.select_next_some() => {
                         if let Some(executed_block) = payload.executed_block() {
-                            debug!(target: "reth::cli", block=?executed_block.block().num_hash(),  "inserting built payload");
-                            eth_service.orchestrator_mut().handler_mut().handler_mut().on_event(EngineApiRequest::InsertExecutedBlock(executed_block).into());
+                            let request = if payload_building_validation_enabled {
+                                debug!(target: "reth::cli", block=?executed_block.block().num_hash(), "inserting built payload with validation");
+                                let (tx, rx) = oneshot::channel();
+                                new_payload_responses.push(rx);
+
+                                let message = BeaconEngineMessage::NewPayload {
```

Comment:
not a fan of this massive select arm, this blows up the entire thing and is hard to parse, this loop was intended to be easy to understand.

this feeds data from one select arm into another, which means this can be converted into a helper type instead